recipes/aws-c-cal/all/conanfile.py

```diff
@@ -98,4 +98,16 @@ def package_info(self):
         elif self.settings.os in ("FreeBSD", "Linux"):
             self.cpp_info.components["aws-c-cal-lib"].system_libs.append("dl")
         if self._needs_openssl:
-            self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
+            if self.options["openssl"].shared:
+                self.cpp_info.components["aws-c-cal-lib"].requires.append("openssl::crypto")
+            else:
+                self.cpp_info.components["_dummy_crypto"].requires.append("openssl::crypto")
```

Comment:
I didn't see that you created a dummy component.
By using `-Wl,-u,XXX`, we need to link to the openssl library.

This might still fail when cmake puts the extra link flags after the openssl library. But let's check first.
```suggestion
                self.cpp_info.components["aws-c-cal-lib"].requires.append("openssl::crypto")
```